plugins/faustwp/tests/integration/settings/test-callbacks.php

```diff
@@ -0,0 +1,190 @@
+		'frontend_uri' => 'http://localhost:3000',
+		'secret_key' => '2b9b7ec9-73f8-407d-b5b3-6f3e56bb555d',
+		'menu_locations' => 'Primary, Footer',
+		'enable_redirects' => '1',
+		'enable_rewrites' => '1',
+		'disable_theme' => '1',
+	];
+
+	public function setUp() {
+		parent::setUp();
+		update_option( $this->option, $this->init_settings );
+	}
+
+	public function test_the_sanitize_option_filter_is_registerd() {
+		$this->assertSame( 10, has_action( 'sanitize_option_faustwp_settings', 'WPE\FaustWP\Settings\sanitize_faustwp_settings' ) );
+	}
+
+	public function test_sanitize_faustwp_settings_does_not_change_existing_settings() {
+		$this->assertSame( $this->init_settings, sanitize_faustwp_settings( $this->init_settings, $this->option ) );
+	}
+
+	public function test_sanitize_faustwp_settings_allows_valid_frontend_uris() {
+		global $wp_settings_errors;
+		// We have to clear this manually since all tests execute in a single "request"
+		$wp_settings_errors = [];
+
+		$valid_uris = [
+			// Clean URLs
+			'' => '',
+			'http://localhost:3000' => 'http://localhost:3000',
+			'https://headless.com/my%20page' => 'https://headless.com/my%20page',
+			'http://localhost:3000/page/?query=headless&page=1' => 'http://localhost:3000/page/?query=headless&page=1',
+			// Dirty URLs
+			'  https://localhost:3000/  ' => 'https://localhost:3000/%20%20',
+			'http://localhost:3000/"^<>{}`' => 'http://localhost:3000/',
+			'https://headless.com?foo[bar]=baz' => 'https://headless.com?foo%5Bbar%5D=baz',
+		];
+
+		foreach ( $valid_uris as $in => $out ) {
+			$updates['frontend_uri'] = $in;
+			$sanitized = sanitize_faustwp_settings( $updates, $this->option );
+
+			$this->assertSame( $out, $sanitized['frontend_uri'] );
+
+			$errors = get_settings_errors( $this->option );
+			$this->assertCount( 0, $errors );
+			// Clear errors between test cases.
+			$wp_settings_errors = [];
+		}
+	}
+
+	public function test_sanitize_faustwp_settings_rejects_invalid_frontend_uris() {
+		global $wp_settings_errors;
+		// We have to clear this manually since all tests execute in a single "request"
+		$wp_settings_errors = [];
+
+		$invalid_uris = [
+			'localhost:3000',
+			'my.example.com',
+			'//example.com',
+			'ftp://files.com',
+			'/headless',
+			'http://',
+		];
+
+		foreach ( $invalid_uris as $url ) {
+			$updates['frontend_uri'] = $url;
+			$sanitized = sanitize_faustwp_settings( $updates, $this->option );
+
+			// Keeps existing value.
+			$this->assertSame( $this->init_settings['frontend_uri'], $sanitized['frontend_uri'] );
+
+			$errors = get_settings_errors( $this->option );
+			$this->assertCount( 1, $errors );
+			$this->assertEquals( $errors[0]['code'], 'faustwp_invalid_frontend_uri' );
+			// Clear errors between test cases.
+			$wp_settings_errors = [];
+		}
+	}
+
+	public function test_sanitize_faustwp_settings_allows_valid_secret_key() {
+		global $wp_settings_errors;
+		// We have to clear this manually since all tests execute in a single "request"
```

Comment:
Maybe move this comment on the top of the file as it is duplicated across the test file.

Reply:
Should be much cleaner in [5e79c55](https://github.com/wpengine/faustjs/pull/782/commits/5e79c55c3f9f2088def907ca9afe924389fdf1eb)